src/Microsoft.IdentityModel.Tokens/TokenValidationResult.cs

```diff
@@ -67,27 +88,73 @@ public ClaimsIdentity ClaimsIdentity
         {
             get
             {
-                if (_claimsIdentity == null)
-                    _claimsIdentity = CreateClaimsIdentity();
+                if (!_claimsIdentityInitialized)
+                {
+                    lock (ClaimsIdentitySyncObj)
+                    {
+                        return ClaimsIdentityNoLocking;
+                    }
+                }
 
                 return _claimsIdentity;
             }
             set
             {
-                _claimsIdentity = value ?? throw LogHelper.LogArgumentNullException(nameof(value));
+                if (value is null)
+                    throw LogHelper.LogArgumentNullException(nameof(value));
+
+                lock (ClaimsIdentitySyncObj)
+                {
+                    ClaimsIdentityNoLocking = value;
+                }
             }
         }
 
         /// <summary>
-        /// This call is for JWTs, SamlTokenHandler will set the ClaimsPrincipal.
+        /// Gets or sets the <see cref="_claimsIdentity"/> without synchronization. All accesses must either
+        /// be protected or used when the caller knows access is serialized.
         /// </summary>
-        /// <returns></returns>
-        private ClaimsIdentity CreateClaimsIdentity()
+        internal ClaimsIdentity ClaimsIdentityNoLocking
```

Comment:
Curious as to the thinking behind this property.

Reply:
ValidateJWEAsync creates a TokenValidationResult and explicitly sets the ClaimsIdentity.  There's no other consumer at that point, and it shouldn't have to pay for the allocation of the lock object or the synchronization on that object in order to set the object.  That same code also accesses the ClaimsIdentity of another instance, which it again knows no one else is currently using, and so it similarly needn't have to lock or allocate.

If that code can go away, ideally because rather than creating a new TokenValidationResult it instead just repopulates the existing one, then you might remove this as well and just inline its contents into ClaimsIdentity.  But as it stands, there are multiple consumers that want the same get/set operations.